### ISSUE TITLE
gn daemon crashing when bidding

### ISSUE THREAD

**Issue by catalyc** [commerceblock/guardnode#18]
Had to ctrl+c and restart the daemon, afterwards I got a 'previously made bid found' - so it seems like it was succesfully sent.

```
2020-02-23 14:07:41,764 Challenge:INFO:1217: Fee address: CP1ke8PCP1fKFkSQzQ7jqgWbfEZn5CvZpD and pubkey: 02193277ab72260435f0145c8$
606c2d20c743f889d8e6c629e255c421f72ebe26                                                                                           
2020-02-23 14:07:41,881 Alerts:INFO:1217: Log file read: /root/dgld/mainnet/ocean/gold_main/debug.log                              
2020-02-23 14:07:41,930 Challenge:INFO:1217: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c$
beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75$
 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('4453.25207427'), 'txid': 'd07a0239b83$
ce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}                                                                             
Exception in thread Thread-1:                                                                                                      
Traceback (most recent call last):                                                                                                 
  File "/root/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 167, in _get_response                                
    http_response = self.__conn.getresponse()                                                                                      
  File "/usr/lib/python3.7/http/client.py", line 1336, in getresponse                                                              
    response.begin()                                                                                                               
  File "/usr/lib/python3.7/http/client.py", line 306, in begin                                                                     
    version, status, reason = self._read_status()                                                                                  
  File "/usr/lib/python3.7/http/client.py", line 267, in _read_status                                                              
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")                                                                       
  File "/usr/lib/python3.7/socket.py", line 589, in readinto                                                                       
    return self._sock.recv_into(b)                                                                                                 
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/root/guardnode/guardnode/challenge.py", line 158, in run
    self.bid_txid = self.bidhandler.do_request_bid(self.request, self.client_fee_pubkey)
  File "/root/guardnode/guardnode/bid.py", line 132, in do_request_bid
    self.service_ocean.importaddress(address)
  File "/root/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 151, in __call__
    response = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/root/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 125, in _request
    return self._get_response()
  File "/root/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 174, in _get_response
    self.__conn.timeout)})
guardnode.qa.tests.test_framework.authproxy.JSONRPCException: 'importaddress' RPC took longer than 30.000000 seconds. Consider usin
g larger timeout for calls that take longer to return. (-344)
```

**Comment by dedekmraz** [comment]
If run_guardnode crashes and needs to be restarted, fee address and pubkey changes. In the second line, when starting, it says "Previously made bid found: txid ... feePubKey ..." with the correct feePubKey of the successful bid. If your new bidlimit is above current auction price and you have enough balance, your main balance gets reduced by current auction price and run_guardnode crashes again. This is true even when there are already enough bids (10) so the auction is practically done. I think it doesn't crash if you specify bidlimit above your balance (would rather not try it) or if you specify bidlimit that is lower than current auction price (but don't know what will happen when auction price decays below your bidlimit). I think it is safe to start it with bidlimit that is below lowest possible auction price (1000 CBT).

**Comment by nkostoulas** [comment]
> If run_guardnode crashes and needs to be restarted, fee address and pubkey changes. In the second line, when starting, it says "Previously made bid found: txid ... feePubKey ..." with the correct feePubKey of the successful bid. If your new bidlimit is above current auction price and you have enough balance, your main balance gets reduced by current auction price and run_guardnode crashes again. This is true even when there are already enough bids (10) so the auction is practically done. I think it doesn't crash if you specify bidlimit above your balance (would rather not try it) or if you specify bidlimit that is lower than current auction price (but don't know what will happen when auction price decays below your bidlimit). I think it is safe to start it with bidlimit that is below lowest possible auction price (1000 CBT).

Quite you provide some logs for this if possible.

**Comment by nkostoulas** [comment]
The balance reduces once you make a bid because the "locked" funds are not included. 

I'd be curious to see why it crashes since the guardnode finds the previous bid.

**Comment by dedekmraz** [comment]
I think it crashes when your bid is "successful" (= lower than the auction price). Until now, I always pressed Ctrl-C when I noticed the error message and restarted GN, usually quite fast after the error. Here are logs from few minutes before successful bid (no errors before auctioPrice>bidlimit) and I also made sure to let it a little longer this time without pressing Ctrl-C, my comments are in <... ...>

2020-02-24 13:19:20,665 Challenge:INFO:7555: Fee address: Ceq3hQLeXg1L244aDSG1J1QwZsi5vSZk7z and pubkey: 03c46f70eb6e41012ebd55a72f5d59edb20043457f40347d275a331ca46bcba0c5
2020-02-24 13:19:20,697 Challenge:INFO:7555: Previously made bid found: {'txid': '4f82a5b69ef226d1da5c239712c44b58bed4a06d56afeda130af3f9d84540072', 'feePubKey': '0355bf5605f3bcc65a8062328ff6665c829a5beac55eb5cd2d460b3b0d69066b1e'}
2020-02-24 13:19:20,698 Alerts:INFO:7555: Log file read: /home/xxx/dgld/gold_main/debug.log
2020-02-24 13:19:20,713 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1110.05593891'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:19:20,729 Challenge:INFO:7555: Request d07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6 ended
2020-02-24 13:19:20,753 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1110.05593891'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:19:20,770 Bid:WARNING:7555: Auction price 1110.05593891 too high for guardnode bid limit 1103.0
2020-02-24 13:20:20,794 Challenge:INFO:7555: No active requests for genesis: c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9
2020-02-24 13:21:20,865 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1108.45015903'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:21:20,883 Bid:WARNING:7555: Auction price 1108.45015903 too high for guardnode bid limit 1103.0
2020-02-24 13:22:20,949 Challenge:INFO:7555: No active requests for genesis: c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9
2020-02-24 13:23:21,006 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1106.84784798'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:23:21,023 Bid:WARNING:7555: Auction price 1106.84784798 too high for guardnode bid limit 1103.0
2020-02-24 13:24:21,060 Challenge:INFO:7555: No active requests for genesis: c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9
2020-02-24 13:25:21,136 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1105.24899581'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:25:21,161 Bid:WARNING:7555: Auction price 1105.24899581 too high for guardnode bid limit 1103.0
2020-02-24 13:26:21,207 Challenge:INFO:7555: No active requests for genesis: c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9
2020-02-24 13:27:21,259 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1103.65359261'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
2020-02-24 13:27:21,275 Bid:WARNING:7555: Auction price 1103.65359261 too high for guardnode bid limit 1103.0
2020-02-24 13:28:21,300 Challenge:INFO:7555: No active requests for genesis: c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9
2020-02-24 13:29:21,385 Challenge:INFO:7555: Found request: {'genesisBlock': 'c66cb6eb7cd585788b294be28c8dcd6be4e37a0a6d238236b11c0beb25833bb9', 'confirmedBlockHeight': 44882, 'startBlockHeight': 47761, 'numTickets': 10, 'decayConst': 83752, 'feePercentage': 75, 'endBlockHeight': 67921, 'startPrice': Decimal('100000.00000000'), 'auctionPrice': Decimal('1102.06162849'), 'txid': 'd07a0239b83fce40f0e1d16517c29b172282669508f484bb69e97fe0ddce79e6'}
<... approx 30 seconds goes by ...>
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/xxxx/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 167, in _get_response
    http_response = self.__conn.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1346, in getresponse
    response.begin()
  File "/usr/lib/python3.6/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.6/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/xxx/guardnode/guardnode/challenge.py", line 158, in run
    self.bid_txid = self.bidhandler.do_request_bid(self.request, self.client_fee_pubkey)
  File "/home/xxx/guardnode/guardnode/bid.py", line 132, in do_request_bid
    self.service_ocean.importaddress(address)
  File "/home/xxx/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 151, in __call__
    response = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/home/xxx/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 125, in _request
    return self._get_response()
  File "/home/xxx/guardnode/guardnode/qa/tests/test_framework/authproxy.py", line 174, in _get_response
    self.__conn.timeout)})
guardnode.qa.tests.test_framework.authproxy.JSONRPCException: 'importaddress' RPC took longer than 30.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
<... I waited approx 5 minutes, there was no other output so I pressed Ctrl-C. Balance decreased (is locked, as you say) for cca 1102 CBT  ...>
^C2020-02-24 13:35:36,728 Guardnode:ERROR:7555: KeyboardInterrupt
